### PR TITLE
Boss spawn enemies

### PR DESCRIPTION
**Pull request by miquelmiro3**
For the task: If combat form: Enemies from lvl 1 & 2 spawns to ensure change weapon to the player triggered by time (for example each 10 seconds to test?)

For testing play the boss scene, after 5 seconds in combat it spawns an enemy (max 2). Let me know if this is what we wanted.

### CODE REVIEW

**Review by monikamnc — APPROVED** [review]
I think for now it's ok ^-^
We can improve it when we have all phases of the boss working.
Great work!

File: Gameplay/entities/enemies/BossController.cpp [hunk 1]
```diff
@@ -441,3 +455,45 @@ void Hachiko::Scripting::BossController::FocusCamera(bool focus_on_boss)
...
+				if (agent)
+				{
+					agent->AddToCrowd();
+				}
+				break;
+			}
+		}
+
+		enemy_timer = 0;
+	}
```
Comment (baransrc) [inline Gameplay/entities/enemies/BossController.cpp:479]:
```suggestion
	if (enemy_timer < time_between_enemies)
	{
            return;
        }

        for (EnemyController* enemy_controller : enemies) 
        {
            GameObject* __restrict enemy = enemy_controller->GetGameObject();

            if (enemy->IsActive())
            {
                continue;
            }

            enemy->SetActive(true);
            ComponentAgent* __restrict agent = enemy->GetComponent<ComponentAgent>();
        
            if (agent)
            {
                agent->AddToCrowd();
            }
            
            break;
        }

        enemy_timer = 0;
```

File: Gameplay/entities/enemies/BossController.cpp [hunk 2]
```diff
@@ -441,3 +455,45 @@ void Hachiko::Scripting::BossController::FocusCamera(bool focus_on_boss)
...
+		}
+		if (enemies[i])
+		{
+			enemies[i]->SetIsFromBoss(true);
+			enemies[i]->ResetEnemy();
+			enemies[i]->ResetEnemyPosition();
+		}
+		enemies[i]->GetGameObject()->SetActive(false);
+	}
+}
```
Comment (baransrc) [inline Gameplay/entities/enemies/BossController.cpp:499]:
```suggestion
void Hachiko::Scripting::BossController::ResetEnemies()
{
	for (EnemyController* enemy_controller : enemies)
	{
        GameObject* __restrict enemy = enemy_controller->GetGameObject();
		ComponentAgent* __restrict agent = enemies[i]->GetGameObject()->GetComponent<ComponentAgent>();
		
        if (agent)
		{
			agent->RemoveFromCrowd();
		}
		
        if (enemy_controller)
		{
            // Maybe pack these methods into a method of EnemyController for
            // ease of extensibility.
			enemy_controller->SetIsFromBoss(true);
			enemy_controller->ResetEnemy();
			enemy_controller->ResetEnemyPosition();
		}
		
        enemy->SetActive(false);
	}
}
```

**Review by baransrc — CHANGES_REQUESTED** [review]
Good work!! There are couple of suggestions that you can address, after that I think we are good to merge!

**Review by Erick9Thor — APPROVED** [review]

**Review by baransrc — APPROVED** [review]

**Pull request merged**